src/main/java/co/fairtide/prometheus_aeron_stat/PrometheusAeronStat.java

```diff
@@ -0,0 +1,263 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package co.fairtide.prometheus_aeron_stat;
+
+import io.aeron.CommonContext;
+import io.aeron.archive.client.AeronArchive;
+import io.aeron.archive.client.RecordingDescriptorConsumer;
+import io.prometheus.client.Counter;
+import io.prometheus.client.Gauge;
+import io.prometheus.client.exporter.HTTPServer;
+import org.agrona.DirectBuffer;
+import org.agrona.IoUtil;
+import org.agrona.concurrent.SigInt;
+import org.agrona.concurrent.status.CountersReader;
+
+import java.io.File;
+import java.nio.MappedByteBuffer;
+import java.nio.charset.StandardCharsets;
+import java.util.HashMap;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+import static io.aeron.CncFileDescriptor.*;
+
+
+
+public class PrometheusAeronStat {
+    /**
+     * The delay in seconds between each update.
+     */
+    private static final String DELAY = "delay";
+    private static final String PROMETHUES_EXPORTER_PORT = "port";
+
+    private CountersReader counters;
+    private HashMap<String, Counter> prometheusCounterMap;
+    private int numChannelPublishingCounter;
+    private int numSubscriberCounter;
+    private int numClientCounter;
+
+    private Gauge numChannelPublishingGauge;
+    private Gauge numSubscriberGauge;
+    private Gauge numClientGauge;
+
+
+    public PrometheusAeronStat(CountersReader counters) {
+        this.counters = counters;
+        this.prometheusCounterMap = new HashMap<>();
+        this.numChannelPublishingGauge = Gauge.build().name("num_channel").help("num_channel").register();
+        this.numSubscriberGauge = Gauge.build().name("num_subscribers").help("num_channel").register();
+        this.numClientGauge = Gauge.build().name("num_clients").help("num_channel").register();
+    }
+
+    public static CountersReader mapCounters() {
+        final File cncFile = CommonContext.newDefaultCncFile();
+        System.out.println("Command `n Control file " + cncFile);
+
+        final MappedByteBuffer cncByteBuffer = IoUtil.mapExistingFile(cncFile, "cnc");
+        final DirectBuffer cncMetaData = createMetaDataBuffer(cncByteBuffer);
+        final int cncVersion = cncMetaData.getInt(cncVersionOffset(0));
+
+        if (CNC_VERSION != cncVersion) {
+            throw new IllegalStateException(
+                    "Aeron CnC version does not match: version=" + cncVersion + " required=" + CNC_VERSION);
+        }
+
+        return new CountersReader(
+                createCountersMetaDataBuffer(cncByteBuffer, cncMetaData),
+                createCountersValuesBuffer(cncByteBuffer, cncMetaData),
+                StandardCharsets.US_ASCII);
+    }
+
+    public static void main(final String[] args) throws Exception {
+        long delayMs = 1000L;
+        int portNumber = -1;
+
+        if (0 != args.length) {
+            checkForHelp(args);
+            for (final String arg : args) {
```

Comment:
Please consider using an external library for command line parsing instead of writing a custom parser. Apache CLI could  be used here
https://commons.apache.org/proper/commons-cli/usage.html

Reply:
Alternatively, you can use Java properties. Please check _io/aeron/samples/SampleConfiguration.java_ for example

Reply:
Changed to apache commons cli